volumina/widgets/layerDialog.py

```diff
@@ -27,146 +27,78 @@
-            if state == 2:
-                self.blueChannelThresholdingWidget.setValue(layer.normalize[2][0], layer.normalize[2][1])  # update gui
-                layer.set_normalize(2, None)  # set to auto
-                self.blueChannelThresholdingWidget.setEnabled(False)
-            if state == 0:
-                self.blueChannelThresholdingWidget.setEnabled(True)
-                layer.set_normalize(2, layer.normalize[2])
-
-        def alphaAutoRange(state):
-            if state == 2:
-                self.alphaChannelThresholdingWidget.setValue(layer.normalize[3][0], layer.normalize[3][1])  # update gui
-                layer.set_normalize(3, None)  # set to auto
-                self.alphaChannelThresholdingWidget.setEnabled(False)
-            if state == 0:
-                self.alphaChannelThresholdingWidget.setEnabled(True)
-                layer.set_normalize(3, layer.normalize[3])
-
-        self.redAutoRange.stateChanged.connect(redAutoRange)
-        self.redAutoRange.setCheckState(layer._autoMinMax[0] * 2)
-        self.greenAutoRange.stateChanged.connect(greenAutoRange)
-        self.greenAutoRange.setCheckState(layer._autoMinMax[1] * 2)
-        self.blueAutoRange.stateChanged.connect(blueAutoRange)
-        self.blueAutoRange.setCheckState(layer._autoMinMax[2] * 2)
-        self.alphaAutoRange.stateChanged.connect(alphaAutoRange)
-        self.alphaAutoRange.setCheckState(layer._autoMinMax[3] * 2)
+        autorange_checkbox.stateChanged.connect(handleAutoRangeChanged)
+        autorange_state = Qt.Checked if self.layer._autoMinMax[datasourceIdx] else Qt.Unchecked
+        autorange_checkbox.setCheckState(autorange_state)
 
-        self.resize(self.minimumSize())
+    def setLayername(self, name: str):
+        self._layerLabel.setText(f"<b>{name}</b>")
 
-    def showRedThresholds(self, show):
-        self.redChannel.setVisible(show)
 
-    def showGreenThresholds(self, show):
-        self.greenChannel.setVisible(show)
+class GrayscaleLayerDialog(LayerDialog):
+    def __init__(self, layer, parent=None):
+        super().__init__(ui_file_name="grayLayerDialog.ui", layer=layer, parent=parent)
 
-    def showBlueThresholds(self, show):
-        self.blueChannel.setVisible(show)
+        self.initialize_range_widgets(
+            datasourceIdx=0,
+            thresholding_widget=self.grayChannelThresholdingWidget,
+            autorange_checkbox=self.grayAutoRange,
+        )
 
-    def showAlphaThresholds(self, show):
-        self.alphaChannel.setVisible(show)
 
-    def setLayername(self, n):
-        self._layerLabel.setText("<b>%s</b>" % n)
+class RGBALayerDialog(LayerDialog):
+    def __init__(self, layer, parent=None):
+        super().__init__(ui_file_name="rgbaLayerDialog.ui", layer=layer, parent=parent)
+
+        for idx, (t_widget, autorange_checkbox, channel) in enumerate(
+            [
+                (self.redChannelThresholdingWidget, self.redAutoRange, self.redChannel),
```

Comment:
I cannot find these functions `redAutoRange`, `greenAutoRange`... anywhere

Reply:
fascinatingly enough it works - no errors related to this.

Reply:
This is the reason I have trust issues with IDE's; they will sometimes miss things that `grep`-like things won't ^^

`redAutoRange`, `greenAutoRange`, etc are checkbox widgets defined at `volumina/widgets/ui/rgbaLayerDialog.ui` and `volumina/widgets/ui/grayLayerDialog.ui` =P

Reply:
ah yeah, I was grepping, too, but only in *.py files :) my miss.!